pkg/crc/machine/bundle/repository.go

```diff
@@ -102,10 +105,23 @@ func (bundle *CrcBundleInfo) createSymlinkOrCopyOpenShiftClient(ocBinDir string)
 }
 
 func (repo *Repository) Extract(path string) error {
-	_, err := extract.Uncompress(path, repo.CacheDir, &extract.Options{
+	dir := filepath.Join(repo.CacheDir, "tmp-extract")
+	_ = os.RemoveAll(dir) // clean up before using it
+	_, err := extract.Uncompress(path, dir, &extract.Options{
 		ShowProgress: true,
+		Flatten:      true,
 	})
-	return err
+	if err != nil {
+		return err
+	}
+	bundle, err := readBundleInfo(dir)
+	if err != nil {
+		return err
+	}
+	if err := bundle.Verify(); err != nil {
+		return err
+	}
+	return os.Rename(dir, filepath.Join(repo.CacheDir, bundle.GetName()))
```

Comment:
> The directory structure of the archive is discarded. The bundle name is built based on the metadata of the bundle.

I don't think it's mandatory to do that in this commit, the bundle directory can be renamed to ` 	path := filepath.Join(repo.CacheDir, strings.TrimSuffix(bundleName, bundleExtension))` as was done before. The use of the bundle name can be done in a separate commit to avoid mixing bug fixes with functional changes.

Reply:
yes good point!